packages/core/strapi/tests/single-type.test.api.js

```diff
@@ -65,6 +66,8 @@ describe('Content Manager single types', () => {
       },
     });
 
+    expect(res.body.data.attributes.publishedAt).toBeISODate();
```

Comment:
Hi, I've pushed the changes and updated the tests. 

The first issue, it's now covered by this line.

For the second issue, no need for additional tests, because the `Update keeps the same data id` is not failing after having `draftAndPublish` set to true ( Before the changes introduced in this pr, it would fail if `draftAndPublish` is true ).